opm/simulators/linalg/cuistl/CuView.hpp

```diff
@@ -0,0 +1,394 @@
+/*
+  Copyright 2022-2023 SINTEF AS
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#ifndef OPM_CUVIEW_HEADER_HPP
+#define OPM_CUVIEW_HEADER_HPP
+#include <dune/common/fvector.hh>
+#include <dune/istl/bvector.hh>
+#include <exception>
+#include <fmt/core.h>
+#include <opm/common/ErrorMacros.hpp>
+#include <opm/simulators/linalg/cuistl/detail/safe_conversion.hpp>
+#include <vector>
+#include <string>
+#include <opm/common/utility/gpuDecorators.hpp>
+
+
+namespace Opm::cuistl
+{
+
+/**
+ * @brief The CuView class is provides a view of some data allocated on the GPU
+ * Essenstially is only stores a pointer and a size.
+ *
+ *  This class supports being used from inside a CUDA/HIP Kernel.
+ *  Implementations are placed in this headerfile for functions that may be called
+ *  inside a kernel to avoid expensive RDC (relocatable device code)
+ *
+ * The view will typically provide a view into a CuBuffer and be able to
+ * manipulate the data within it
+ *
+ * @param T Type of the data we store, typically int/float/double w/o const specifier
+ *
+ **/
+template <typename T>
+struct CuView
```

Comment:
Should be a `class` and should have its values (`m_dataPtr` etc etc) private.

Reply:
done